test/configCases/parsing/issue-6867/test.js

```diff
@@ -0,0 +1,3 @@
+export default (function hello () {
+	return 1;
+})
```

Comment:
I'd feel better if there were tests for each permutation.

```js
// function _declaration_, id is null, in native implementations the function is named "default"
export default function() {}
```

```js
// function _expression_, id is null, in native implementations the function is named "default"
export default (function() {})
```

```js
// function _declaration_, id is "foo"
export default function foo() {
  foo = function() {} // this not working is currently a bug in webpack 😝
})
```

```js
// function _expression_
export default (function bar() {})
```

Reply:
These same permutations also apply to `class` declarations, but we probably don't need to test them here 🤔